### PR TITLE
Adapt to changes in OpenML API, bump versions, add maxbytes feature

### PR DESCRIPTION
**Pull request by jbrea**
This addresses #16 and #17.

In addition,
- I refactored the handling of GET requests to simplify code maintenance.
- I adapted to changes in the OpenML API
- I added the keyword argument `maxbytes` to the `load` function to support loading of the first few rows of datasets.
- I removed the keyword `api_keys`, which didn't have any effect so far unless making some functions throw errors (this is technically breaking).

### CODE REVIEW

**Comment by ablaom** [comment]
Many thanks @jbrea !

@darenasc or @OkonSamuel  Going on leave for a week. Any chance one of you could review this?

Hopefully I can then jump and tag a new (breaking release) with the new HTTP compat.

**Comment by jbrea** [comment]
Does anyone have the time to look at this within the next few days? I would like to use the updated version for the course that starts mid September.

File: Project.toml [hunk 1]
```diff
@@ -11,8 +11,8 @@ Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
 Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 
 [compat]
-ARFFFiles = "1.3"
-HTTP = "0.8, 0.9"
+ARFFFiles = "^1.4.1"
```
Comment (ablaom) [inline Project.toml]: In Julia project files the caret `^` is understood. 

```suggestion
ARFFFiles = "1.4.1"
```

File: src/data.jl [hunk 2]
```diff
@@ -6,38 +6,47 @@ const API_URL = "https://www.openml.org/api/v1/json"
...
-        end
+        r = HTTP.request("GET", string(API_URL, url))
+        return JSON.parse(String(r.body))
     catch e
-        println("Error occurred. Check if there exists a dataset with id $id.")
-        println("See e.g. OpenML.list_datasets()\n")
-        println(e)
-        return nothing
+        error_msg_handling(e)
+        extra_error_message != "" && println(extra_error_message)
```
Comment (ablaom) [inline src/data.jl]: We never actually throw an exception, only print error messages. Is there a good reason we don't (eventually) throw the exception?

**Review by ablaom — COMMENTED** [review]
Thanks for this contribution @jbrea . 

You have marked this as a breaking release. Can you briefly itemize what is breaking for the public API, so we can record this is the release notes?

File: src/data.jl (hunk 2 above)
Comment (jbrea) [reply src/data.jl]: No. I fixed this. Thanks.

**Comment by jbrea** [comment]
Breaking is only the removal of the keyword argument `api_keys`. However, this keyword argument never did anything, so I don't expect to break user code.

File: src/data.jl (hunk 2 above)
Comment (ablaom) [reply src/data.jl]: Yes but it looks like this is a premature throw. Don't you want the message at line 32 of data.jl to also be printed?

Comment (jbrea) [reply src/data.jl]: You're right. I changed this now to throw all errors. The error messages look somewhat intimidating, but I guess this is the proper way of doing it.

**Review by ablaom — APPROVED** [review]
Good to go.

**Comment by jbrea** [comment]
Thanks, @ablaom!

**Pull request merged**